flake.nix

```diff
@@ -0,0 +1,187 @@
+{
+  description = "Build a cargo project";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    crane.url = "github:ipetkov/crane";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.rust-analyzer-src.follows = "";
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+
+    advisory-db = {
+      url = "github:rustsec/advisory-db";
+      flake = false;
+    };
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    crane,
+    fenix,
+    flake-utils,
+    advisory-db,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = nixpkgs.legacyPackages.${system};
+
+      inherit (pkgs) lib;
+      toolchain = fenix.packages.${system}.fromToolchainFile {
+        file = ./rust-toolchain.toml;
+        sha256 = "sha256-3jVIIf5XPnUU1CRaTyAiO0XHVbJl12MSx3eucTXCjtE=";
+      };
+
+      craneLib = (crane.mkLib pkgs).overrideToolchain toolchain;
+      src = craneLib.cleanCargoSource ./.;
+
+      # Common arguments can be set here to avoid repeating them later
+      commonArgs = {
+        inherit src;
+        strictDeps = true;
+
+        buildInputs = with pkgs;
+          [
+            libxkbcommon
+            xorg.libX11
+            vulkan-loader
+            vulkan-validation-layers
+            vulkan-headers
+          ]
+          ++ lib.optionals pkgs.stdenv.isDarwin [
+            # Additional darwin specific inputs can be set here
+            pkgs.libiconv
+          ];
+
+        nativeBuildInputs = with pkgs; [copyDesktopItems pkg-config makeWrapper];
+
+        # Additional environment variables can be set directly
+        # MY_CUSTOM_VAR = "some value";
+      };
+
+      craneLibLLvmTools =
+        craneLib.overrideToolchain
+        (toolchain.withComponents [
+          "cargo"
+          "llvm-tools"
+          "rustc"
+        ]);
+
+      # Build *just* the cargo dependencies, so we can reuse
+      # all of that work (e.g. via cachix) when running in CI
+      cargoArtifacts = craneLib.buildDepsOnly commonArgs;
+
+      # Build the actual crate itself, reusing the dependency
+      # artifacts from above.
+      scope = craneLib.buildPackage (commonArgs
+        // {
+          inherit cargoArtifacts;
+          desktopItems = [
+            (pkgs.makeDesktopItem {
+              name = "Scope";
+              desktopName = "Scope";
+              exec = "scope";
+              icon = "Scope";
+              terminal = false;
+              # TODO: keywords
+              keywords = [];
+              startupWMClass = "Scope";
+            })
+          ];
+
+          postInstall = let
+            icon = pkgs.fetchurl {
+              url = "https://avatars.githubusercontent.com/u/188419158?s=200&v=4";
```

Comment:
Perhaps we should use an avatar hosted somewhere more permanently

Reply:
github avatars are pretty much permanent, updates even if you change your profile picture.

Reply:
Good to know, thanks @vmfunc